tensorflow/compiler/xla/service/hlo_instruction.cc

```diff
@@ -3606,6 +3609,14 @@ const string& HloInstruction::custom_call_target() const {
   return Cast<HloCustomCallInstruction>(this)->custom_call_target();
 }
 
+const bool HloInstruction::has_side_effect() const {
```

Comment:
Oh, we can't do this, having a function `HloInstruction::has_side_effect()` that only applies to custom-call would be extremely confusing, especially since we already have `HloInstruction::HasSideEffect`.

It should only be exposed on HloCustomCallInstruction.  And since we already have HasSideEffect, maybe we don't even need `has_side_effect`, even on CustomCall and can get away just with the setter (which should also only be on the custom-call class).

Reply:
Yeah I wasn't sure about that, other fields have getters/setters through HloInstruction but it would be confusing. I've removed these and added casts.

Reply:
Yeah, we're not consistent about it, many fields have them but not all.  If we really wanted a getter/setter on HloInstruction we'd want to call it custom_call_side_effect or something.  But I think as-is is better anyway.